test/show.jl

```diff
@@ -248,6 +248,8 @@ end
 @test repr(:(∓ 1)) == ":(∓1)"
 @test repr(:(± 1)) == ":(±1)"
 
+eval_parse_repr(x) = eval(Meta.parse(repr(x)))
+eval_parse_repr_test(x) = @test eval_parse_repr(x) == x
```

Comment:
This makes the line numbers reported by a test failure worse. 

```
julia> using Test

julia> @test 1==2
Test Failed at REPL[2]:1 <---- Just entered
  Expression: 1 == 2
   Evaluated: 1 == 2

ERROR: There was an error during testing

julia> @test 1==3
Test Failed at REPL[3]:1 <---- Just entered
  Expression: 1 == 3
   Evaluated: 1 == 3

ERROR: There was an error during testing

julia> test_eq_1(x) = @test x == 1
test_eq_1 (generic function with 1 method)

julia> test_eq_1(2)
Test Failed at REPL[4]:1 <----- where function defined
  Expression: x == 1
   Evaluated: 2 == 1

ERROR: There was an error during testing

julia> test_eq_1(3)
Test Failed at REPL[4]:1 <----- where function defined
  Expression: x == 1
   Evaluated: 3 == 1

ERROR: There was an error during testing
```

Reply:
So just make it return a bool and add back the `@test`.

Reply:
We can have either line numbers or display the actual/expected values. I prefer displaying the actual/expected values (this) because we can always get line numbers from stack traces.

Reply:
What do you mean? As is, this is just worse than before. Finding the line number in the stacktrace is often not easy.

Reply:
> What do you mean?

```julia
julia> f(x) = x == 5
f (generic function with 1 method)

julia> @test f(4)
Test Failed at REPL[15]:1
  Expression: f(4)

ERROR: There was an error during testing

julia> g(x) = @test x == 5
g (generic function with 1 method)

julia> g(4)
Test Failed at REPL[16]:1
  Expression: x == 5
   Evaluated: 4 == 5

ERROR: There was an error during testing
```